platform-sdk/swirlds-common/src/main/java/com/swirlds/common/io/filesystem/FileSystemManager.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright (C) 2024 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.swirlds.common.io.filesystem;
+
+import com.swirlds.base.state.Startable;
+import com.swirlds.base.state.Stoppable;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import edu.umd.cs.findbugs.annotations.Nullable;
+import java.io.IOException;
+import java.nio.file.Path;
+
+/**
+ * Responsible for organizing and managing access to the file system.
+ */
+public interface FileSystemManager extends Startable, Stoppable {
+
+    /**
+     * Resolve a path relative to the root directory of the file system manager.
+     * Implementations can choose the convenient subfolder inside the root directory.
+     *
+     * @param relativePath the path to resolve against the root directory
+     * @return the resolved path
+     * @throws IllegalArgumentException if the path is "below" the root directory (e.g. resolve("../foo")
+     */
+    @NonNull
+    Path resolve(@NonNull Path relativePath);
+
+    /**
+     * Creates a path relative to the root directory of the file system manager.
+     * Implementations can choose the convenient subfolder inside the root directory.
+     * There is no file or directory actually being created after the invocation of this method.
+     *
+     * @param tag if indicated, will be suffixed to the returned path
+     * @return the resolved path
+     * @throws IllegalArgumentException if the path is "below" the root directory (e.g. resolve("../foo")
```

Comment:
I tend to think of the example being "above" the root directory

Reply:
Agree